packages/next-swc/crates/next-core/src/next_server/context.rs

```diff
@@ -128,22 +128,40 @@ pub async fn get_server_resolve_options_context(
     let invalid_styled_jsx_client_only_resolve_plugin =
         get_invalid_styled_jsx_resolve_plugin(project_path);
 
+    let mut transpile_packages = next_config.transpile_packages().await?.clone_value();
+    transpile_packages.extend(
+        (*next_config.optimize_package_imports().await?)
+            .iter()
+            .cloned(),
+    );
+
     // Always load these predefined packages as external.
     let mut external_packages: Vec<String> = load_next_js_templateon(
         project_path,
         "dist/lib/server-external-packages.json".to_string(),
     )
     .await?;
 
-    let transpile_packages = next_config.transpile_packages().await?;
-    external_packages.retain(|item| !transpile_packages.contains(item));
+    let server_external_packages = &*next_config.server_external_packages().await?;
+
+    let conflicting_packages = transpile_packages
+        .iter()
+        .filter(|package| server_external_packages.contains(package))
+        .cloned()
```

Comment:
Not sure if this `cloned()` is necessary

Reply:
Maybe not, but it will only allocate if there are conflicts